modules/programs/mise.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mise;
+  tomlFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ hm.maintainers.pedorich-n ];
+
+  imports = let
+    mkRemovedWarning = opt:
+      (mkRemovedOptionModule [ "programs" "rtx" opt ] ''
+        The `rtx` package has been replaced by `mise`, please switch over to
+        using the options under `programs.mise.*` instead.
+      '');
+
+  in map mkRemovedWarning [
+    "enable"
+    "package"
+    "enableBashIntegration"
+    "enableZshIntegration"
+    "enableFishIntegration"
```

Comment:
These all seem like they can be `mkRenamedOptionModule`, and just make `settings` be a `mkRemovedOptionModule`

Reply:
That would be cool. Perhaps best not to rename `package`, though since I guess it is likely to point to an rtx package in existing configurations?

Reply:
Given that the configuration from `rtx` doesn't translate fully to `mise` and `settings` now means a different thing I would argue that it's better to fail fast and let the user know they must take action. 

I don't think we should allow configuration like this to be valid

```nix
    rtx = {
      enable = true;
      enableFishIntegration = false;
    };

    mise = {
      settings = {
        experimental = true;
      };
      globalConfig = {
        tools = {
          python = [ "3.11" ];
        };
      };
    };
```